File: ai-service-python/app/models/fund_nav_daily.py
from datetime import date
from decimal import Decimal

from sqlalchemy import Date, Numeric, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, UUIDMixin


class FundNavDaily(UUIDMixin, Base):
    __tablename__ = "fund_nav_daily"

    fund_id: Mapped[str] = mapped_column(UUID(as_uuid=True), ForeignKey("funds.id", ondelete="CASCADE"), nullable=False)
    nav_date: Mapped[date] = mapped_column(Date, nullable=False)
    unit_nav: Mapped[Decimal] = mapped_column(Numeric(12, 4), nullable=False)
    acc_nav: Mapped[Decimal | None] = mapped_column(Numeric(12, 4))
    daily_return: Mapped[Decimal | None] = mapped_column(Numeric(8, 4))
    total_return: Mapped[Decimal | None] = mapped_column(Numeric(8, 4))

File: backend-go/internal/model/fund_nav_daily.go
package model

import (
	"time"

	"github.com/shopspring/decimal"
)

// FundNavDaily 每日净值模型
type FundNavDaily struct {
	ID          string          `gorm:"type:uuid;primaryKey;default:uuid_generate_v4()" json:"id"`
	FundID      string          `gorm:"type:uuid;index" json:"fund_id"`
	NavDate     time.Time       `gorm:"type:date" json:"nav_date"`
	UnitNav     decimal.Decimal `gorm:"type:numeric(12,4)" json:"unit_nav"`
	AccNav      *decimal.Decimal `gorm:"type:numeric(12,4)" json:"acc_nav,omitempty"`
	DailyReturn *decimal.Decimal `gorm:"type:numeric(8,4)" json:"daily_return,omitempty"`
	TotalReturn *decimal.Decimal `gorm:"type:numeric(8,4)" json:"total_return,omitempty"`
	CreatedAt   time.Time       `json:"created_at"`

	Fund Fund `gorm:"foreignKey:FundID" json:"fund,omitempty"`
}

func (FundNavDaily) TableName() string { return "fund_nav_daily" }
